File: webserver/src/main/java/com/llh/webserver/controller/AuthController.java
package com.llh.webserver.controller;

import cn.hutool.core.util.StrUtil;
import com.llh.webserver.common.exception.BusinessException;
import com.llh.webserver.common.exception.msg.BasicExpEnum;
import com.llh.webserver.pojo.JsonWrapper;
import com.llh.webserver.pojo.auth.AuthTokenVO;
import com.llh.webserver.pojo.auth.LoginVO;
import com.llh.webserver.pojo.auth.RegisterOrUpdateVO;
import com.llh.webserver.service.AuthService;
import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.validation.Valid;

/**
 * AuthController
 * <p>
 * CreatedAt: 2020-04-18 17:47
 *
 * @author llh
 */
@RestController
@Api("授权服务")
@RequestMapping("auth")
@Slf4j
public class AuthController extends BasicController {
    @Autowired
    @Qualifier("authService")
    private AuthService authService;

    @PostMapping("register")
    @ApiOperation("注册新用户")
    public JsonWrapper register(@RequestBody RegisterOrUpdateVO registerUserVO) {
        if (StrUtil.isBlank(registerUserVO.getPassword())) {
            log.debug("注册用户时密码不能为空！ {}", registerUserVO);
            throw new BusinessException(BasicExpEnum.DATA_VALIDATE_ERROR);
        }
        authService.register(registerUserVO);
        return JsonWrapper.ok(registerUserVO);
    }

    @PostMapping("login")
    @ApiOperation("登录操作")
    public JsonWrapper login(@RequestBody @Valid LoginVO loginVO) {
        AuthTokenVO tokenVO = authService.login(loginVO);
        return JsonWrapper.ok(tokenVO);
    }
}

File: server4kt/src/main/kotlin/com/llh/server/controller/AuthController.kt
package com.llh.server.controller

import com.llh.server.pojo.JsonWrapper
import com.llh.server.pojo.okResponse
import com.llh.server.pojo.vo.LoginVO
import com.llh.server.service.sys.AuthService
import io.swagger.annotations.Api
import io.swagger.annotations.ApiOperation
import org.apache.logging.log4j.kotlin.Logging
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import javax.validation.Valid

/**
 * AuthController 登录登出操作的控制器
 *
 * CreatedAt: 2020-06-11 21:58
 *
 * @author llh
 */
@RestController
@RequestMapping("auth")
@Api("授权服务")
class AuthController : Logging {

    @Autowired
    @Qualifier("authService")
    private lateinit var authService: AuthService

    @PostMapping("login")
    @ApiOperation("登录操作")
    fun login(@RequestBody @Valid loginVO: LoginVO): JsonWrapper {
        val tokenVO = authService.login(loginVO)
        return okResponse(tokenVO)
    }
}
